src/Query.php

```diff
@@ -14,6 +14,58 @@ class Query
     protected $type;
     protected $wheres = [];
     protected $options = [];
+    protected $cacheQueryResults = false;
+    protected static $cachedRecords = [];
+
+    public static function eagerLoadAll(): Query
+    {
+        (new static)->cache()->get();
+
+        return new static;
+    }
+
+    public static function setRecordCache(Collection $records): Collection
+    {
+        return static::$cachedRecords = $records->keyBy('id');
+    }
+
+    public static function clearRecordCache(): Collection
+    {
+        return static::$cachedRecords = collect();
+    }
+
+    public static function addToRecordCache(Collection $records): Collection
```

Comment:
Add or Push?

Reply:
I think add is okay in this instance, it'd been renamed to `addToDocumentCache($documents)`